### PR TITLE
Adding support to footer links in mailers

### PR DESCRIPTION
**Pull request by tdg**
<a
href="https://www.swapmyvote.uk/?utm_source=email&utm_medium=welcome_ema
il&utm_campaign=site">Swap My Vote</a>
%br
<a
href="https://twitter.com/intent/follow?screen_name=swapmyvote">Follow
us on Twitter</a>
%br
<a href="https://facebook.com/swapmyvote">Like us on Facebook</a>
%br
<a href="https://crowdfunder.co.uk/swapmyvote">Support us on
Crowdfunder</a>

### CODE REVIEW

File: app/views/user_mailer/no_swap.haml [hunk 1]
```diff
@@ -5,16 +5,18 @@
...
+  If a progressive or pro-EU vote is what is important to you, <a href="http://tacticalvote.co.uk">TacticalVote.co.uk</> has been very popular this election. We are not aware of any similar site on the right or Brexit sides of the political compass.
 %p
   Thank you very much for taking part - together we are making democracy better and more representative.
 %p
   All the very best,
   %br
   Swap My Vote Team
 %p
-  <a href="https://www.swapmyvote.uk/?utm_source=email&utm_medium=no_swap&utm_campaign=site">Swap My Vote</a>
+  <a href="https://www.swapmyvote.uk/?utm_source=email&utm_medium=welcome_email&utm_campaign=site">Swap My Vote</a>
```
Comment (jpallen) [inline app/views/user_mailer/no_swap.haml]: Wrong `utm_medium`

File: app/views/user_mailer/not_swapped_follow_up.haml [hunk 2]
```diff
@@ -16,10 +16,10 @@
 	%br
 	Swap My Vote Team
 %p
-	<a href="https://www.swapmyvote.uk/?utm_source=email&utm_medium=not_swapped_follow_up&utm_campaign=site">Swap My Vote</a>
+	<a href="https://www.swapmyvote.uk/?utm_source=email&utm_medium=welcome_email&utm_campaign=site">Swap My Vote</a>
```
Comment (jpallen) [inline app/views/user_mailer/not_swapped_follow_up.haml]: Likewise

File: app/views/user_mailer/partner_has_voted.html.haml [hunk 3]
```diff
@@ -16,9 +16,10 @@
   %br
   Swap My Vote Team
 %p
-  <a href="https://www.swapmyvote.uk/?utm_source=email&utm_medium=partner_has_voted&utm_campaign=site">Swap My Vote</a>
+  <a href="https://www.swapmyvote.uk/?utm_source=email&utm_medium=welcome_email&utm_campaign=site">Swap My Vote</a>
```
Comment (jpallen) [inline app/views/user_mailer/partner_has_voted.html.haml]: Ditto

File: app/views/user_mailer/reminder_to_vote.html.haml [hunk 4]
```diff
@@ -14,8 +14,10 @@
   %br
   Swap My Vote Team
 %p
-  <a href="https://www.swapmyvote.uk/?utm_source=email&utm_medium=reminder_to_vote&utm_campaign=site">Swap My Vote</a>
+  <a href="https://www.swapmyvote.uk/?utm_source=email&utm_medium=welcome_email&utm_campaign=site">Swap My Vote</a>
```
Comment (jpallen) [inline app/views/user_mailer/reminder_to_vote.html.haml]: Same again

File: app/views/user_mailer/swap_cancelled.haml [hunk 5]
```diff
@@ -12,8 +12,10 @@
 	%br
 	Swap My Vote Team
 %p
-	<a href="https://www.swapmyvote.uk/?utm_source=email&utm_medium=swap_cancelled&utm_campaign=site">Swap My Vote</a>
+	<a href="https://www.swapmyvote.uk/?utm_source=email&utm_medium=welcome_email&utm_campaign=site">Swap My Vote</a>
```
[the remark this answers is not in the document]
Comment (jpallen) [reply app/views/user_mailer/swap_cancelled.haml]: And here!

File: app/views/user_mailer/swap_confirmed.html.haml [hunk 6]
```diff
@@ -22,8 +22,10 @@
 	%br
 	Swap My Vote Team
 %p
-	<a href="https://www.swapmyvote.uk/?utm_source=email&utm_medium=swap_confirmed&utm_campaign=site">Swap My Vote</a>
+	<a href="https://www.swapmyvote.uk/?utm_source=email&utm_medium=welcome_email&utm_campaign=site">Swap My Vote</a>
```
[the remark this answers is not in the document]
Comment (jpallen) [reply app/views/user_mailer/swap_confirmed.html.haml]: Here too!

File: app/views/user_mailer/swap_not_confirmed.html.haml [hunk 7]
```diff
@@ -10,8 +10,10 @@
   %br
   Swap My Vote Team
 %p
-  <a href="https://www.swapmyvote.uk/?utm_source=email&utm_medium=swap_not_confirmed&utm_campaign=site">Swap My Vote</a>
+  <a href="https://www.swapmyvote.uk/?utm_source=email&utm_medium=welcome_email&utm_campaign=site">Swap My Vote</a>
```
[the remark this answers is not in the document]
Comment (jpallen) [reply app/views/user_mailer/swap_not_confirmed.html.haml]: Another one!

**Comment by aspiers** [comment]
Only glanced at this but it sounds like there is an uncomfortable amount of duplication of HAML going on - please can we start using partials to fix this?  Although probably best to merge this and fix duplication in a follow-up.

**Pull request merged**